src/callbacks_step/stepsize_dg2d.jl

```diff
@@ -159,6 +159,22 @@ function max_dt(u, t, mesh::ParallelP4estMesh{2},
     return dt
 end
 
+function max_dt(u, t, mesh::ParallelT8codeMesh{2},
+                constant_speed::False, equations, dg::DG, cache)
+    # call the method accepting a general `mesh::T8codeMesh{2}`
+    # TODO: MPI, we should improve this; maybe we should dispatch on `u`
+    #       and create some MPI array type, overloading broadcasting and mapreduce etc.
+    #       Then, this specific array type should also work well with DiffEq etc.
+    dt = invoke(max_dt,
+                Tuple{typeof(u), typeof(t), T8codeMesh{2},
+                      typeof(constant_speed), typeof(equations), typeof(dg),
+                      typeof(cache)},
+                u, t, mesh, constant_speed, equations, dg, cache)
+    dt = MPI.Allreduce!(Ref(dt), min, mpi_comm())[]
+
+    return dt
+end
+
 function max_dt(u, t, mesh::ParallelP4estMesh{2},
                 constant_speed::True, equations, dg::DG, cache)
```

Comment:
Unrelated to this PR, but I am wondering: Why do we actually need to dispatch on `constant_speed`? Looks like both `max_dt` functions are the same. Couldn't we just get rid of one of the methods and don't dispatch on `constant_speed`?

Reply:
Thanks for the hint! I would agree.

Reply:
Tried it.

Reply:
Well, it does not work but I do not get the point. 

The error is
```
LoadError: MethodError: max_dt(...) is ambiguous.

 Candidates:
    max_dt(u, t, mesh::Union{Trixi.P4estMesh{3}, Trixi.StructuredMesh{3}, Trixi.T8codeMesh{3}}, constant_speed::Static.False, equations, dg::Trixi.DG, cache)
      @ Trixi ~/work/Trixi.jl/Trixi.jl/src/callbacks_step/stepsize_dg3d.jl:47
    max_dt(u, t, mesh::Union{Trixi.P4estMesh{3, <:Real, <:Static.True}, Trixi.T8codeMesh{3, <:Real, <:Static.True}}, constant_speed, equations, dg::Trixi.DG, cache)
      @ Trixi ~/work/Trixi.jl/Trixi.jl/src/callbacks_step/stepsize_dg3d.jl:122
  
  Possible fix, define
    max_dt(::Any, ::Any, ::Union{Trixi.P4estMesh{3, <:Real, <:Static.True}, Trixi.T8codeMesh{3, <:Real, <:Static.True}}, ::Static.False, ::Any, ::Trixi.DG, ::Any)
```

Reply:
Probably just keep it like it is now, I guess.

Reply:
So we would need to change `<:Static.True` to `<:Static.False` in a generic way to work both for `T8codeMesh` and `P4estMesh`...

Reply:
Currently reverted to what it was like before.